### ISSUE TITLE
Currupts Resource.db

### ISSUE THREAD

**Issue by hgdagon** [zeffy/panyi#2]
This corrupts the file in a way that some strings show up in English, however most stay in Chinese.
That said I tried replacing manually with openmcdf viewer with the exact same result.

**Comment by zeffy** [comment]
I haven't looked at this in a while, but iirc it was always like that. Some of the application strings just aren't in `StringTable.xml`, so panyi can only get it part of the way there.

**Comment by hgdagon** [comment]
Well, I've since figured that out on my own.However it's not hardcoded, it's  the skins zip file. I personally don't have that much time, but translating that zip file is much more important (and a headache on it's own, since if you don't use a specific program, Baidu can't read the file and crashes), since there's a lot more text in there, than in the compound file.
However, even though those strings not being translated isn't related to the file, the file is still being corrupted.

On the left is my db file, on the right is the one generated by panyi. Look at `<Property name="limit" type="int"/>`. It's in the end of the file in both the original file, and mine. Panyi moves it to the the middle of the file (line 3231). I generated mine with StructuredStorageXplorer, adding directories and files one by one.
![corrupt](https://user-images.githubusercontent.com/15173076/71668875-1cc7a380-2d84-11ea-91fb-f4eb5a1c1983.png)

**Comment by zeffy** [comment]
Interesting, maybe it is an issue with OpenMcdf? I didn't encounter this when I originally made panyi, so I'm not sure.

If I have some time I will try to look into this and additionally see if I can expose a way to translate other files contained in Resource.db to make panyi more extendable. Thanks.

**Comment by schecteracademicservices** [comment]
Updated panyi.ini and DuiStringTable.xml (goes in the duiengineskin.zip file in the xml folder) for version 6.8.9.1 generated using google translate.

just delete the ".txt" and enjoy

[DuiStringTable.xml.txt](https://github.com/zeffy/panyi/files/4021452/DuiStringTable.xml.txt)
[panyi.ini.txt](https://github.com/zeffy/panyi/files/4021451/panyi.ini.txt)

**Comment by hgdagon** [comment]
Well, I guess, it woyldn't hurt to post mine as well. This is based on a previous translation by another guy (a native speaker, I think). I reworked it a little, fixed some grammatical errors, and translated new lines missing from his translations with Google Translate (don't speak Chinese myself). If I had a tool that could edit both the compound file and the zip file (in v6.8.4.1 editing the zip file with WinRAR or 7-zip resulted in a crash) and some more time at hand I could probably spend more time on this. In the mean time, if you stumbled upon this by accident, compare mine with the one uploaded by @schecteracademicservices and see which one you're most comfortable with,
[Baidu Cloud 6.8.4.1 EN.zip](https://github.com/zeffy/panyi/files/4022843/Baidu.Cloud.6.8.4.1.EN.zip)

**Comment by schecteracademicservices** [comment]
Would you please post your strings file (panyi.ini) for resources.db? It's easier to parse. Thx!

**Comment by Mark-Joy** [comment]
Thank you all for the files!!! :)

**Comment by schecteracademicservices** [comment]
hgdagon's translations are better, but are missing a few new additions.  I've updated it - attached.
[panyi - best of.ini.txt](https://github.com/zeffy/panyi/files/4047269/panyi.-.best.of.ini.txt)

**Comment by hgdagon** [comment]
@schecteracademicservices Yeah, I think Baidu got updated since, I never had time to add any of the new strings. Also, I ended up not using Panyi, so I don't have an ini file to share. However getting an ini file is as easy as extracting the xml (i.e. with 7-Zip) and few quick operations in Notepad++.
